### PR TITLE
AND-3725 | Implement missed call notification and relavant event

### PR DESCRIPTION
**Pull request by hchauhansf**
In this PR, we have implemented both the feature of missed call notification and related events & disconnect incoming call from notification and full screen.

Attached both the tickets here,

https://phonecom.atlassian.net/browse/AND-3725
https://phonecom.atlassian.net/browse/AND-3726

### CODE REVIEW

**Comment by pratik-pdc** [comment]
@SirSaunders @dheerajkatarya No action required. Just added, so you can get the updates on current progress.

File: sipservice/src/main/java/net/gotev/sipservice/BroadcastEventEmitter.java [hunk 1]
```diff
@@ -302,6 +302,28 @@ public void sendCallMediaEvent(int mediaEventType) {
...
     }
 
+    public void errorCallback(String message) {
+        final Intent intent = new Intent();
+        intent.setAction(getAction(BroadcastAction.CALLBACK_GENERIC_ERROR));
+        intent.putExtra(PARAM_ERROR_MESSAGE, message);
+        sendExplicitBroadcast(intent);
+    }
+
+    public void handleMissedCall(boolean isIncomingCall, String number, String linkedUUid,
```
Comment (pratik-pdc) [inline sipservice/src/main/java/net/gotev/sipservice/BroadcastEventEmitter.java:312]: Please check on naming convention.

File: sipservice/src/main/java/net/gotev/sipservice/BroadcastEventEmitter.java [hunk 2]
```diff
@@ -302,6 +302,28 @@ public void sendCallMediaEvent(int mediaEventType) {
...
+                                 String callerName, long time, long seconds, CallType callType) {
+        Intent intent = new Intent();
+        intent.putExtra(PARAM_IS_INCOMING_CALL, isIncomingCall);
+        intent.putExtra(PARAM_PHONE_NUMBER, number);
+        intent.putExtra(PARAM_INCOMING_LINKED_UUID, linkedUUid);
+        intent.putExtra(PARAM_CALLER_NAME, callerName);
```
Comment (pratik-pdc) [inline sipservice/src/main/java/net/gotev/sipservice/BroadcastEventEmitter.java:318]: Use "PARAM_DISPLAY_NAME"

File: sipservice/src/main/java/net/gotev/sipservice/SipAccount.java [hunk 3]
```diff
@@ -86,7 +87,7 @@ public SipCall addOutgoingCall(final String numberToDial, boolean isVideo, boole
 
         // check if there's already an ongoing call
         int totalCalls = 0;
-        for (SipAccount _sipAccount: SipService.getActiveSipAccounts().values()) {
+        for (SipAccount _sipAccount : SipService.getActiveSipAccounts().values()) {
```
Comment (pratik-pdc) [inline sipservice/src/main/java/net/gotev/sipservice/SipAccount.java:90]: Please check usage of this snippet.

File: sipservice/src/main/java/net/gotev/sipservice/SipCall.java [hunk 4]
```diff
@@ -274,14 +274,15 @@ public void sendBusyHereToIncomingCall() {
         }
     }
 
-    public void declineIncomingCall() {
+    public void declineIncomingCall() throws Exception{
```
Comment (pratik-pdc) [inline sipservice/src/main/java/net/gotev/sipservice/SipCall.java:277]: Check on unused methods..

File: sipservice/src/main/java/org/pjsip/pjsua2/extras/CallbackMessageConst.java [hunk 5]
```diff
@@ -0,0 +1,5 @@
+package org.pjsip.pjsua2.extras;
+
+public class CallbackMessageConst {
+    public static final String ERR_SIP_ACCOUNT_NULL = "Sip account in not found.";
```
Comment (pratik-pdc) [inline sipservice/src/main/java/org/pjsip/pjsua2/extras/CallbackMessageConst.java:4]: Move this constant to SipServiceConstant

**Review by pratik-pdc — APPROVED** [review]
Approved! Please accommodate comments.

File: sipservice/src/main/java/net/gotev/sipservice/BroadcastEventEmitter.java (hunk 1 above)
Comment (hchauhansf) [reply sipservice/src/main/java/net/gotev/sipservice/BroadcastEventEmitter.java:312]: All changes will be accommodated in AND-3730 PR.

File: sipservice/src/main/java/net/gotev/sipservice/BroadcastEventEmitter.java (hunk 2 above)
Comment (hchauhansf) [reply sipservice/src/main/java/net/gotev/sipservice/BroadcastEventEmitter.java:318]: All changes will be accommodated in AND-3730 PR.

File: sipservice/src/main/java/net/gotev/sipservice/SipAccount.java (hunk 3 above)
Comment (hchauhansf) [reply sipservice/src/main/java/net/gotev/sipservice/SipAccount.java:90]: All changes will be accommodated in AND-3730 PR.

File: sipservice/src/main/java/net/gotev/sipservice/SipCall.java (hunk 4 above)
Comment (hchauhansf) [reply sipservice/src/main/java/net/gotev/sipservice/SipCall.java:277]: All changes will be accommodated in AND-3730 PR.

File: sipservice/src/main/java/org/pjsip/pjsua2/extras/CallbackMessageConst.java (hunk 5 above)
Comment (hchauhansf) [reply sipservice/src/main/java/org/pjsip/pjsua2/extras/CallbackMessageConst.java:4]: All changes will be accommodated in AND-3730 PR.

**Pull request merged**